front/app/resources/GetProjectAllowedInputTopics.tsx

```diff
@@ -22,51 +12,9 @@ type Children = (
   renderProps: IProjectAllowedInputTopicsState
 ) => JSX.Element | null;
 
-export default class GetProjectAllowedInputTopics extends React.Component<
-  Props,
-  State
-> {
-  private subscription: Subscription;
-
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      projectAllowedInputTopics: undefined,
-    };
-  }
-
-  setProjectAllowedInputTopics(
-    projectAllowedInputTopics: IProjectAllowedInputTopicsState
-  ) {
-    this.setState({ projectAllowedInputTopics });
-  }
+const GetProjectAllowedInputTopics = ({ children, projectId }: Props) => {
+  const projectAllowedInputTopics = useProjectAllowedInputTopics(projectId);
```

Comment:
Using the resource hook instead of having the same code (subscriptions etc.) in both the resource hook and resource component.

Reply:
Tested in places where it's used and works as before.